assets/js/blocks/product-collection/inspector-controls/columns-control.tsx

```diff
@@ -12,44 +13,86 @@ import {
 /**
  * Internal dependencies
  */
-import { DisplayLayoutControlProps } from '../types';
+import { DisplayLayoutToolbarProps } from '../types';
 import { getDefaultDisplayLayout } from '../constants';
 
-const ColumnsControl = ( props: DisplayLayoutControlProps ) => {
-	const { type, columns } = props.displayLayout;
+const toggleLabel = __(
+	'Shrink columns to fit',
+	'woo-gutenberg-products-block'
+);
+
+const toggleHelp = __(
+	'Reduce the number of columns to better fit smaller screens and spaces.',
+	'woo-gutenberg-products-block'
+);
+
+const getColumnsLabel = ( shrinkColumns: boolean ) =>
+	shrinkColumns
+		? __( 'Max Columns', 'woo-gutenberg-products-block' )
+		: __( 'Columns', 'woo-gutenberg-products-block' );
+
+const ColumnsControl = ( props: DisplayLayoutToolbarProps ) => {
+	const { type, columns, shrinkColumns } = props.displayLayout;
 	const showColumnsControl = type === 'flex';
 
 	const defaultLayout = getDefaultDisplayLayout();
 
+	const onToggleChange = ( value: boolean ) => {
+		props.setAttributes( {
+			displayLayout: {
+				...props.displayLayout,
+				shrinkColumns: value,
+			},
+		} );
+	};
+
+	const onPanelDeselect = () => {
+		props.setAttributes( {
+			displayLayout: defaultLayout,
+		} );
+	};
+
+	const onColumnsChange = ( value: number ) =>
+		props.setAttributes( {
+			displayLayout: {
+				...props.displayLayout,
+				columns: value,
+			},
+		} );
+
 	return showColumnsControl ? (
-		<ToolsPanelItem
-			label={ __( 'Columns', 'woo-gutenberg-products-block' ) }
-			hasValue={ () =>
-				defaultLayout?.columns !== columns ||
-				defaultLayout?.type !== type
-			}
-			isShownByDefault
-			onDeselect={ () => {
-				props.setAttributes( {
-					displayLayout: defaultLayout,
-				} );
-			} }
-		>
-			<RangeControl
-				label={ __( 'Columns', 'woo-gutenberg-products-block' ) }
-				value={ columns }
-				onChange={ ( value: number ) =>
-					props.setAttributes( {
-						displayLayout: {
-							...props.displayLayout,
-							columns: value,
-						},
-					} )
+		<>
+			<ToolsPanelItem
+				hasValue={ () =>
+					defaultLayout?.shrinkColumns !== shrinkColumns
+				}
+				isShownByDefault
+				onDeselect={ onPanelDeselect }
+			>
+				<ToggleControl
+					checked={ !! shrinkColumns }
+					label={ toggleLabel }
+					help={ toggleHelp }
+					onChange={ onToggleChange }
```

Comment:
I would suggest a more descriptive naming for the "onChange" callbacks because if we ever want to add more "toggle" controls, `onToggleChange` would not describe which settings it is referring to without looking into the details. Perhaps `onShrinkColToggleChange`?

Reply:
Addressed in https://github.com/woocommerce/woocommerce-blocks/pull/11320/commits/99b6415905451fefeba480bd347be6d3bfd17ee2 👍